client-js/app/elements/labrad-manager.ts

```diff
@@ -28,14 +40,14 @@ export class LabradManager extends polymer.Base {
 
     this.mgr.connectionClose(id);
     this.selectedConnection = null;
+    this.updateConnections();
+  }
 
-    // Remove the connection from the array to notify the UI
-    const numConnections = this.connections.length;
-    for (let i = 0; i < numConnections; ++i) {
-      if (this.connections[i].id == id) {
-        this.splice('connections', i, 1);
-        break;
-      }
+  private async updateConnections(): Promise<void> {
+    const connections = await this.mgr.connections();
+    this.splice('connections', 0, this.connections.length);
+    for (let c of connections) {
```

Comment:
Nice! I think there are a few other places we splice that could use this trick too.